dust-lambda-extension/extensions/dust

```diff
@@ -0,0 +1,294 @@
+			return 1
+		fi
+
+		if ! chmod +x "$TMP_CHALK_PATH"; then
+			LOG_ERROR "Failed to make chalk binary executable in /tmp"
+			return 1
+		fi
+	fi
+
+	return 0
+}
+
+# Process chalk marks with better error handling
+process_chalk_marks() {
+	LOG_INFO "$TMP_CHALK_PATH extracting $CHALK_MARK_PATH"
+	if [[ -f $CHALK_MARK_PATH ]]; then
+		LOG_DEBUG "Found $CHALK_MARK_PATH, running chalk extract."
+		if ! "$TMP_CHALK_PATH" extract "$CHALK_MARK_PATH"; then
+			LOG_WARNING "chalk extract command failed"
+			return 1
+		fi
+	else
+		LOG_WARNING "No chalk.json found at $CHALK_MARK_PATH"
+		return 1
+	fi
+
+	return 0
+}
+
+# JSON parsing using pure bash/standard tools (no jq dependency)
+extract_json_field() {
+	local json="$1"
+	local field="$2"
+
+	# Use grep with improved regex to handle various JSON formatting
+	local result
+	result=$(echo "$json" | grep -o "\"$field\"[[:space:]]*:[[:space:]]*\"[^\"]*\"" |
+		sed "s/\"$field\"[[:space:]]*:[[:space:]]*\"\([^\"]*\)\"/\1/")
+
+	# Handle case where field might have null value
+	if [[ $result == "null" ]]; then
+		result=""
+	fi
+
+	echo "$result"
+}
+
+extract_extension_id() {
+	local temp_file="$1"
+	# Extract the Extension ID from response headers
+	EXTENSION_ID=$(awk '/^[Ll]ambda-[Ee]xtension-[Ii]dentifier:/ {
+        gsub(/\r$/, "");
+        gsub(/^[^:]*:[[:space:]]*/, "");
+        print $0;
+        exit
+    }' "$temp_file")
+
+	rm -f "$temp_file"
+
+	if [[ -z $EXTENSION_ID ]]; then
+		LOG_ERROR "Failed to register extension or parse Extension ID."
+		exit 1
+	fi
+}
+
+register_extension() {
+	# Only register for SHUTDOWN events (INVOKE events are not needed)
+	local registration_payload='{"events":["SHUTDOWN"]}'
+
+	# Make a single registration request with headers included
+	local temp_file
+	temp_file=$(mktemp)
+	local http_code
+
+	http_code=$(curl --silent --write-out "%{http_code}" \
+		--dump-header "$temp_file" \
+		--request POST "$REGISTER_API" \
+		--header "Lambda-Extension-Name: $EXTENSION_NAME" \
+		--header "Content-Type: application/json" \
+		--data "$registration_payload" \
+		--output /dev/null)
+
+	if [[ $http_code -ne 200 ]]; then
+		LOG_ERROR "Failed to register extension. HTTP status: $http_code"
+		LOG_DEBUG "Response headers: $(cat "$temp_file")"
+		rm -f "$temp_file"
+		exit 1
+	fi
+
+	extract_extension_id "$temp_file"
+
+	LOG_INFO "Extension registered with ID: $EXTENSION_ID"
+}
+
+# Main execution starts here
+main() {
+	LOG_INFO "Starting Lambda extension: $EXTENSION_NAME"
+
+	# Set up signal handlers
+	trap handle_signal SIGINT SIGTERM
+
+	register_extension
+
+	# Setup chalk binary
+	if ! setup_chalk_binary; then
+		LOG_ERROR "Failed to setup chalk binary"
+		exit 1
+	fi
+
+	# Process chalk marks
+	process_chalk_marks || LOG_WARNING "Chalk mark processing failed, continuing anyway"
+
+	# Poll for events from the Lambda Runtime API
+	LOG_INFO "Starting event polling loop."
+
+	local consecutive_failures=0
+	local max_failures=$MAX_FAILURES
```

Comment:
why alias local vars to exactly copy of a global var?

Reply:
I typically take a local copy for functions to keep state encapsulated. I'll dedupe.